tests/rust_tests.rs

```diff
@@ -345,12 +415,130 @@ fn requirements_test() {
             sc.require_minting_is_ready();
         })
         .assert_ok();
+
+    b_wrapper
+        .execute_query(&setup.contract_wrapper, |sc| {
+            sc.require_is_privileged(&managed_address!(owner_address))
+        })
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &owner_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                sc.set_administrator(managed_address!(first_user_address));
+            },
+        )
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &first_user_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                sc.set_max_supply(managed_biguint!(20));
+            },
+        )
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &first_user_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                sc.set_is_paused(true);
+            },
+        )
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &second_user_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                sc.set_is_paused(true);
+            },
+        )
+        .assert_user_error("Address is not privileged");
+
+    b_wrapper
+        .execute_query(&setup.contract_wrapper, |sc| {
+            sc.require_is_privileged(&managed_address!(second_user_address))
+        })
+        .assert_user_error("Address is not privileged");
+
+    b_wrapper
+        .execute_query(&setup.contract_wrapper, |sc| {
+            sc.require_value_is_positive(&managed_biguint!(0u64));
+        })
+        .assert_error(4, "Value must be positive");

```

Comment:
Error returned by that function should be changed to "Value must be higher than or equal to zero" if it accepts 0, too. Value must be positive can be misleading